ui-spacetimechart/src/utils/canvas.ts

```diff
@@ -273,3 +273,39 @@ export function drawPathExtremity(
     drawPathStopExtremity(ctx, timePixel, spacePixel, swapAxis);
   }
 }
+
+/**
+ *
+ * @param minT number timestamp
+ * @param maxT number timestamp
+ * @param timeRanges time frames (24h, 12h, 6h, …)
+ * @param gridlinesLevels width of the lines for each time frame
+ * @returns Record<number, number>
+ * Keys are times in ms
+ * Values are the highest level on each time
+ */
+
+export function displayElementsBasedOnZoom<T>(
+  minT: number,
+  maxT: number,
+  timeRanges: number[],
+  levels: number[],
+  formatter: (level: number, i: number) => T = identity
```

Comment:
I'd prefer to apply this as a second step to reduce the complexity of that function. (The docs seem outdated.)

Reply:
we have to use Object.entries / re map over the object / Object.fromEntries
this seems way more cumbersome than passing a simple formatter

Reply:
`mapValues` from lodash makes this less cumbersome, FWIW.

(But I'm not feeling strongly about this.)

Reply:
think i’m gonna stick with the original idea, `mapValues` doesn’t keep the indices